plugin/pkg/admission/noderestriction/admission.go

```diff
@@ -603,6 +640,71 @@ func (p *Plugin) admitServiceAccount(nodeName string, a admission.Attributes) er
 	return nil
 }
 
+func (p *Plugin) podReferencesAudience(pod *v1.Pod, audience string) bool {
+	for _, v := range pod.Spec.Volumes {
+		if v.Projected != nil {
+			for _, src := range v.Projected.Sources {
+				if src.ServiceAccountToken != nil && src.ServiceAccountToken.Audience == audience {
+					return true
+				}
+			}
+		}
+
```

Comment:
would it be better or worse to maintain the set of audiences that *all* CSI drivers have in their token requests and only descend into volumesource -> {pvc -> pv} -> CSI driver traversal madness if the audience could *possibly* be in that set?